packages/application-generic/src/services/auth/shared.ts

```diff
@@ -0,0 +1,54 @@
+export const buildOauthRedirectUrl = (request): string => {
+  if (!request.user || !request.user.token) {
+    return `${
+      process.env.FRONT_BASE_URL + '/auth/login'
+    }?error=AuthenticationError`;
+  }
+
+  let url = process.env.FRONT_BASE_URL + '/auth/login';
+  const redirectUrl = JSON.parse(request.query.state).redirectUrl;
+
+  /**
+   * Make sure we only allow localhost redirects for CLI use and our own success route
+   * https://github.com/novuhq/novu/security/code-scanning/3
+   */
+  if (
+    redirectUrl &&
+    redirectUrl.startsWith('http://localhost:') &&
+    !redirectUrl.includes('@')
+  ) {
+    url = redirectUrl;
+  }
+
+  url += `?token=${request.user.token}`;
+
+  if (request.user.newUser) {
+    url += '&newUser=true';
+  }
+
+  /**
+   * partnerCode, next and configurationId are required during external partners integration
+   * such as vercel integration etc
+   */
+  const partnerCode = JSON.parse(request.query.state).partnerCode;
+  if (partnerCode) {
+    url += `&code=${partnerCode}`;
+  }
+
+  const next = JSON.parse(request.query.state).next;
+  if (next) {
+    url += `&next=${next}`;
+  }
+
+  const configurationId = JSON.parse(request.query.state).configurationId;
+  if (configurationId) {
+    url += `&configurationId=${configurationId}`;
+  }
+
+  const invitationToken = JSON.parse(request.query.state).invitationToken;
+  if (invitationToken) {
+    url += `&invitationToken=${invitationToken}`;
+  }
```

Comment:
```suggestion
  const parsedQueryState = JSON.parse(request.query.state);
  const { configurationId, invitationToken, next, partnerCode } = parsedQueryState;
  if (partnerCode) {
    url += `&code=${partnerCode}`;
  }

  if (next) {
    url += `&next=${next}`;
  }

  if (configurationId) {
    url += `&configurationId=${configurationId}`;
  }

  if (invitationToken) {
    url += `&invitationToken=${invitationToken}`;
  }
```
And this could be isolated in a function that handles all these conditions to build app to the URL.